src/domains/invariant.ml

```diff
@@ -1,6 +1,38 @@
 open Cil
 
-type t = exp option
+module ExpLat =
+struct
+  include CilType.Exp
+
+  let bot () = zero (* false *)
+  let top () = one (* true *)
+  let is_bot _ = failwith "ExpLat: is_bot"
+  let is_top _ = failwith "ExpLat: is_top"
+
+  let leq _ _ = failwith "ExpLat: leq"
+  let pretty_diff () _ = failwith "ExpLat: pretty_diff"
+
+  let join x y = BinOp (LOr, x, y, intType)
+  let meet x y = BinOp (LAnd, x, y, intType)
```

Comment:
Should there not be some type of simplification here? Or at least a handling of joining `x` and `x` to yield `x` instead of `x || x` or meeting `x` and `x` leading to `x` instead of `x&x`?

Reply:
I suppose there could, but this is more or less how the invariant combination happened before and that hasn't been an issue since we don't have multiple analyses outputting syntactically identical invariants. Even if that were the case, in any nontrivial case AC-rewriting would be needed to avoid such redundancy.

Since these operations are fully syntactic and `leq` isn't even implemented (nor needed), the lack of idempotence, etc shouldn't be a problem in any way. But for future reference I'll add a couple of comments here about this.